kotlinx-coroutines-debug/src/internal/Attach.kt

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2016-2020 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+@file:Suppress("unused")
+package kotlinx.coroutines.debug.internal
+
+import net.bytebuddy.*
+import net.bytebuddy.agent.*
+import net.bytebuddy.dynamic.loading.*
+
+/*
+ * These classes are used reflectively from kotlinx-coroutines-core when this module is present in the classpath.
+ * It is a substitute for service loading
+ */
+internal class ByteBuddyDynamicAttach : Function0<Unit> {
```

Comment:
Separate classes for attach&detach look quite excessive. It also forces an extra code in the `core` module (a separate `getFunction`, lots of string class names, etc). You can have a single `class DynamicAttach : (Boolean) -> Unit` with a `Boolean` parameter that means `attach` or `detach` operation. Should make the code smaller.